### PR TITLE
Assignment 7 - Part 1 + 2 - Oliver222

### PR DESCRIPTION
**Pull request by Oliver-222**
>build.gradle: Added necessary plugins, dependencies, and protobuf support
>Created coordinator_service.proto
>Created an computationCoordinatorClientInput and Output file (.txt) used for testing the program and services (I used when debugging)

______

>ComputeEngine: Added executeJob for CoordinatorService.ComputeRequest
>ComputeEngineImpl: Changed ArrayList to List so it is compatible with the CoordinatorService.ComputeRequest request.getFactorsList() and implemented executeJob
>FactorAPI: Added basic message (was missing)
>GetFactorResponseImpl: Added errorMessage
>ReadRequest: Added ReadRequest(String source) to get source for CoordinatorService without needing computeRequest
>WriteRequest: Added a constructor to support CoordinatorService.ComputeRequest
>ComputeEngineComputationTest: Fixed typo

### CODE REVIEW

**Comment by Oliver-222** [comment]
FIXED THIS ISSUE:

It needs to be noted that the checkstyle is failing only off of the auto-generated files. We need to figure out how we can make it so checkstyle doesn't check these (2) files.

File: test/ComputeEngineComputationTest.java [hunk 1]
```diff
@@ -20,7 +20,7 @@ public static void testComputeEngine(){
                     break;
                 }
                     if(tempValue < 0) {
-                        throw new IllegalArgumentException("nvalid input. Please enter a positive integer.");
```
Comment (mirahsupreme) [inline test/ComputeEngineComputationTest.java:23]: just a mispelling here but it looks good It wouldn't let me edit

**Review by mirahsupreme — APPROVED** [review]
Well-organized gRPC method descriptors and stub factories, making the gRPC service straightforward to use with both async and blocking options. 

The protobuf plugin setup is clear and correctly specifies the versions for gRPC and protobuf which makes it straightforward to update or modify versions if needed in the future. everything looks good

File: test/ComputeEngineComputationTest.java (hunk 1 above)
Comment (Oliver-222) [reply test/ComputeEngineComputationTest.java:23]: typo was fixed

**Pull request merged**